include/thread_queue.h

```diff
@@ -1,19 +1,39 @@
 #pragma once
 
+#include "affinity.h"
 #include "async_event.h"
 #include "double_buffered_queue.h"
 #include "named_threads.h"
 #include "tracy.h"
 #include "utils.h"
 
 #include <chrono>
+#include <concepts>
+#include <exception>
 #include <future>
 #include <thread>
 #include <type_traits>
 #include <variant>
 
 namespace celerity::detail {
 
+// A helper to implement the body of a worker thread while taking care of all boilerplate code
+void thread_body(const named_threads::thread_type t_type, const std::invocable auto& fun) {
+	thread_pinning::name_and_pin_thread(t_type);
+	const auto name = named_threads::thread_type_to_string(t_type);
+	CELERITY_DETAIL_TRACY_SET_THREAD_NAME_AND_ORDER(t_type);
+	try {
+		fun();
+	}
+	// LCOV_EXCL_START
+	catch(std::exception& e) { //
+		utils::panic("exception in {}: {}", name, e.what());
+	} catch(...) { //
+		utils::panic("exception in {}", name);
+	}
```

Comment:
It was probably me who came up with this pattern, but I'm having second thoughts. There's one or two hard to reproduce bugs in the runtime right now (`std::get wrong index for variant`) which produce a crash dump that shows the `panic` as the reason of fault instead of the stack trace for the exception causing it. If we let the exception terminate the program, we would get a proper trace.

Maybe a try-catch would be better suited around invocations of user code such as host tasks.

Reply:
That's a good point, maybe we should just not catch it and let the program fail?
(In the distant future we could use standardized mechanisms to get a proper stack trace...)

Reply:
We decided offline that for debugging purposes the best thing to do is to just let the default handler happen in all circumstances.